packages/front-end/components/Layout/Layout.tsx

```diff
@@ -386,7 +386,8 @@ const Layout = (): React.ReactElement => {
 
   const [upgradeModal, setUpgradeModal] = useState(false);
   const showUpgradeButton =
-    ["oss", "starter"].includes(accountPlan || "") ||
+    (["oss", "starter"].includes(accountPlan || "") &&
+      !organization.isVercelIntegration) ||
```

Comment:
50/50 on this change. On one hand, think hiding this could be beneficial, since they can't actually do it (they just get instructed to go to Vercel).

Reply:
I would probably not do this change.  There are several places in the app we tell people to upgrade, but we don't link to the upgrade modal directly and rely on that Upgrade button always being in the left nav.